### PR TITLE
HDDS-7271. Ozone Integration test shows memory leak (graceful shutdown cleanup)

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

1. Cleanup for RatisDropwizardExports registry on stop/shutdown
2. Avoiding continuous loop after interrupt for DeleteBlocksCommandHandler

3. Other services not stopped after stopping the cluster is also handled.

Few cases for "ForkJoinPool" which is based on CompletablFuture can not be handled as part, as this depends on service logic for handling same and marking it close.

These is observed in Ozone integration test

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7271

## How was this patch tested?

This is verified running ozone integration test and verifying heap dump for same. This issue is not observed.

### CODE REVIEW

**Comment by sumitagrawl** [comment]
@ChenSammi @nandakumar131 Please review

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java
Comment (ChenSammi) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java:62]: Hey @szetszwo, could you help to review this Ratis metrics un-register logic ?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java
Comment (ChenSammi) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:232]: Please keep the origin implementation.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/BackgroundSCMService.java
Comment (ChenSammi) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/BackgroundSCMService.java:129]: Why remove this statement?

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sumitagrawl for working on this.  Please avoid reordering imports unnecessarily.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/BackgroundSCMService.java
Comment (sumitagrawl) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/BackgroundSCMService.java:129]: When current stopping thread calls interrupt, its interrupted and further handling is stopped throwing interrupt exception. This is causing to stop to fail.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java
Comment (sumitagrawl) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:232]: When call stop of serviceManager, as its registered part of serviceManager causes stop to fail. We must support stop as common framework or do nothing in stop.
So this is removed.

Comment (adoroszlai) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:232]: In other words it's not an unsupported operation, rather a no-op.

`BlockManagerImpl` explicitly starts and shuts down `SCMBlockDeletingService`, which is also registered with `SCMServiceManager`, which manages start/stop of registered services.  I think it's confusing to have both of these lifecycle management calls for the same service.  Can we merge `stop` and `shutdown` and let only `SCMServiceManager` manage it?

Comment (sumitagrawl) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:232]: 1. For this PR, will avoid serviceManager.stop() and scope out this cleanup
2. OR remove serviceManager.stop() and add explicit member fields of class and do cleanup as SCM stop
3. OR SCMBlockDeletingService stop() call internally shutdown()
Which process should we do currently?

Comment (sumitagrawl) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java:232]: I checked, BlockManagerImpl is also calling close of blockingService twice (may be code bug but not in scope). But this has no impact.
Considering this, will call shutdown() in stop() of SCMBlockDeletingService, and this do not have any impact for code reuse.

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java
Comment (szetszwo) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java:62]: +1 the change looks good.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
@ChenSammi please wait for clean CI before merging PRs.  It seems two integration tests are timing out due to this change:

```
org.apache.hadoop.ozone.dn.ratis.TestDnRatisLogParser
org.apache.hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion
```

**Comment by adoroszlai** [comment]
I have reverted the change, please fix the two failing tests and open new PR.

**Pull request merged**
